obspy/io/sac/sactrace.py

```diff
@@ -1105,12 +1102,15 @@ def read(cls, source, headonly=False, ascii=False, byteorder=None,
 
         .. rubric:: Example
 
-        >>> sac = SACTrace.read(filename, headonly=True) # doctest: +SKIP
-        >>> try: # doctest: +SKIP
-                sac.validate('data_hdrs') # doctest: +SKIP
-            except SacInvalidContentError: # doctest: +SKIP
-                sac = SACTrace.read(filename, headonly=False) # doctest: +SKIP
-                sac.validate('data_hdrs') # doctest: +SKIP
+        >>> from obspy.core.util import get_example_file
+        >>> from obspy.io.sac.util import SacInvalidContentError
+        >>> file_ = get_example_file("test.sac")
+        >>> sac = SACTrace.read(file_, headonly=True)
+        >>> try:
+        ...     sac.validate('data_hdrs')
+        ... except (ValueError, SacInvalidContentError):
+        ...     sac = SACTrace.read(file_, headonly=False)
+        ...     sac.validate('data_hdrs')
```

Comment:
I don't think that having files with incorrect data headers is a use case we intended to support. I'm for fixing the header and dealing with the fallout.

________________________________
From: Tobias Megies <notifications@github.com>
Sent: Sunday, June 12, 2016 3:44:52 AM
To: obspy/obspy
Cc: Maccarthy, Jonathan K; Mention
Subject: Re: [obspy/obspy] Better SACTrace docs (#1430)


In obspy/io/sac/sactrace.py<https://github.com/obspy/obspy/pull/1430#discussion_r66718676>:

> @@ -1105,12 +1102,15 @@ def read(cls, source, headonly=False, ascii=False, byteorder=None,
>
>          .. rubric:: Example
>
> -        >>> sac = SACTrace.read(filename, headonly=True) # doctest: +SKIP
> -        >>> try: # doctest: +SKIP
> -                sac.validate('data_hdrs') # doctest: +SKIP
> -            except SacInvalidContentError: # doctest: +SKIP
> -                sac = SACTrace.read(filename, headonly=False) # doctest: +SKIP
> -                sac.validate('data_hdrs') # doctest: +SKIP
> +        >>> from obspy.core.util import get_example_file
> +        >>> from obspy.io.sac.util import SacInvalidContentError
> +        >>> file_ = get_example_file("test.sac")
> +        >>> sac = SACTrace.read(file_, headonly=True)
> +        >>> try:
> +        ...     sac.validate('data_hdrs')
> +        ... except (ValueError, SacInvalidContentError):
> +        ...     sac = SACTrace.read(file_, headonly=False)
> +        ...     sac.validate('data_hdrs')


Dunno.. maybe just fixing the header (i.e. setting with actual mean) in the test file would make sense? (Unless that makes othere tests fail.. ;-)

-
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub<https://github.com/obspy/obspy/pull/1430/files/191e0f962da542ccb0820ec1353c557c873953b1..7f74e34c04252b3d0ed18b3b562100cd820b3070#r66718676>, or mute the thread<https://github.com/notifications/unsubscribe/AGLNkEvRvK0r1YbWhOXwjWRXjYxFDENVks5qK9UUgaJpZM4Ip5Th>.